addons/project/static/src/project_sharing/views/list/list_renderer.js

```diff
@@ -1,34 +0,0 @@
-/** @odoo-module */
-
-import { evaluateBooleanExpr } from "@web/core/py_js/py";
-import { ListRenderer } from "@web/views/list/list_renderer";
-import { onWillUpdateProps } from "@odoo/owl";
-
-export class ProjectSharingListRenderer extends ListRenderer {
-    setup() {
-        super.setup(...arguments);
-        this.setColumns(this.allColumns);
-        onWillUpdateProps((nextProps) => {
-            this.setColumns(nextProps.archInfo.columns);
-        });
-    }
-
-    setColumns(columns) {
-        if (this.props.list.records.length) {
-            const allColumns = [];
-            const firstRecord = this.props.list.records[0];
-            for (const column of columns) {
-                if (evaluateBooleanExpr(column.column_invisible, firstRecord.evalContextWithVirtualIds)) {
-                    continue;
-                }
-                allColumns.push(column);
-            }
-            this.allColumns = allColumns;
-        } else {
-            this.allColumns = columns;
-        }
-        this.state.columns = this.allColumns.filter(
-            (col) => !col.optional || this.optionalActiveFields[col.name]
-        );
-    }
```

Comment:
We should deprecate that method instead because some custom code could override that method. 🙂

Reply:
Something like this ?

 ```js
 export class ProjectSharingListRenderer extends ListRenderer {
    setColumns(columns) {
        this.allColumns = columns;
        this.state.columns = this.allColumns.filter(
            (col) => !col.optional || this.optionalActiveFields[col.name]
        );
    }
}
```